DocumentDataAPI/DocumentDataAPI/Data/Repositories/NpgWordRatioRepository.cs

```diff
@@ -46,20 +46,24 @@ public int Add(WordRatioModel entity)
             });
     }
 
-    public int AddBatch(List<WordRatioModel> entities)
+    public async Task<int> AddBatch(List<WordRatioModel> entities)
     {
         int rowsAffected = 0;
         using IDbConnection con = _connectionFactory.CreateConnection();
         con.Open();
         using IDbTransaction transaction = con.BeginTransaction();
         try
         {
-            foreach (WordRatioModel[] chunk in entities.Chunk(_sqlHelper.InsertStatementChunkSize))
+            var tasks = entities.Chunk(_sqlHelper.InsertStatementChunkSize).Select(async chunk =>
             {
-                string parameterString = _sqlHelper.GetBatchInsertParameters(chunk, out Dictionary<string, dynamic> parameters);
-                rowsAffected += transaction.Execute(
-                    "insert into word_ratios(documents_id, word, amount, percent, rank) values " + parameterString, parameters);
-            }
+                string parameterString =
+                    _sqlHelper.GetBatchInsertParameters(chunk, out Dictionary<string, dynamic> parameters);
+                return await transaction.ExecuteAsync(
+                    "insert into word_ratios(documents_id, word, amount, percent, rank) values "
+                    + parameterString, parameters);
+            });
+            var result = await Task.WhenAll(tasks);
+            rowsAffected = result.Sum();
```

Comment:
Har testet dette - det medfører at Dapper eksekverer flere af disse inserts på samme tid, hvilket den ikke kan, og så thrower den en exception. 
Vi kan godt gøre selve metoden async, men vi er nødt til at lave de her inserts sekventielt, så dette vil f.eks. se sådan ud:
```csharp
foreach (DocumentModel[] chunk in models.Chunk(_sqlHelper.InsertStatementChunkSize))
{
    string parameterString = _sqlHelper.GetBatchInsertParameters(chunk, out Dictionary<string, dynamic> parameters);
    rowsAffected += await transaction.ExecuteAsync("insert into documents(id, sources_id, title, path, summary, date, author, total_words) values " + parameterString, parameters);
}
```

Reply:
Det samme gælder for den tilsvarende metode i de andre repositories